pkg/apis/internal.admin.acorn.io/v1/computeclassresources_test.go

```diff
@@ -0,0 +1,454 @@
+package v1
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestComputeClassResourcesAdd(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		name     string
+		current  ComputeClassResources
+		incoming ComputeClassResources
+		expected ComputeClassResources
+	}{
+		{
+			name:    "add to empty ComputeClassResources resources",
+			current: ComputeClassResources{},
+			incoming: ComputeClassResources{"foo": {
+				Memory: resource.MustParse("1Mi"),
+			}},
+			expected: ComputeClassResources{"foo": {
+				Memory: resource.MustParse("1Mi"),
+			}},
+		},
+		{
+			name: "add to existing ComputeClassResources resources",
+			current: ComputeClassResources{"foo": {
+				Memory: resource.MustParse("1Mi"),
+			}},
+			incoming: ComputeClassResources{"foo": {
+				Memory: resource.MustParse("1Mi"),
+			}},
+			expected: ComputeClassResources{"foo": {
+				Memory: resource.MustParse("2Mi"),
+			}},
+		},
+		{
+			name: "add where current has a resource specified with unlimited",
+			current: ComputeClassResources{"foo": {
+				Memory: UnlimitedQuantity(),
+			}},
+			incoming: ComputeClassResources{"foo": {
+				Memory: resource.MustParse("1Mi"),
+			}},
+			expected: ComputeClassResources{"foo": {
+				Memory: UnlimitedQuantity(),
+			}},
+		},
+		{
+			name: "add where incoming has a resource specified with unlimited",
+			current: ComputeClassResources{"foo": {
+				Memory: resource.MustParse("1Mi"),
+			}},
+			incoming: ComputeClassResources{"foo": {
+				Memory: UnlimitedQuantity(),
+			}},
+			expected: ComputeClassResources{"foo": {
+				Memory: UnlimitedQuantity(),
+			}},
+		},
+		{
+			name: "add where current and incoming have a resource specified with unlimited",
+			current: ComputeClassResources{"foo": {
+				Memory: UnlimitedQuantity(),
+			}},
+			incoming: ComputeClassResources{"foo": {
+				Memory: UnlimitedQuantity(),
+			}},
+			expected: ComputeClassResources{"foo": {
+				Memory: UnlimitedQuantity(),
+			}},
+		},
+		{
+			name: "add where current and incoming have AllComputeClasses specified at non-unlimited values",
+			current: ComputeClassResources{AllComputeClasses: {
+				Memory: resource.MustParse("1Mi"),
+			}},
+			incoming: ComputeClassResources{AllComputeClasses: {
+				Memory: resource.MustParse("1Mi"),
+			}},
+			expected: ComputeClassResources{AllComputeClasses: {
+				Memory: resource.MustParse("2Mi"),
+			}},
+		},
+	}
+
+	// Run the test cases
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.current.Add(tc.incoming)
+			assert.True(t, tc.current.Equals(tc.expected))
+		})
+	}
+}
+
+func TestComputeClassResourcesRemove(t *testing.T) {
```

Comment:
Can we add a test case here for the suspected issue?

Current has two compute classes, one is removed, and the other should still be there.

You kind of have something like this here, but the difference is that the remove touches both compute classes in the included test case. I am suggesting that the other compute class is not touched by the test case.